src/effects/VST/VSTEffect.cpp

```diff
@@ -1695,13 +1695,94 @@ bool VSTEffect::LoadCommon()
 
 #endif
 
+   AEffect* tempHandle;
+
+   try
+   {
+      tempHandle = mPluginMain(VSTEffectWrapper::AudioMaster);
+   }
+   catch (...)
+   {
+      wxLogMessage(wxT("VST plugin initialization failed\n"));
+      return false;
+   }
+
+   // Ask the plugin to identify itself...might be needed for older plugins
+   callDispatcher(tempHandle, effIdentify, 0, 0, NULL, 0);
+
+   // Open a temporary plugin handle, to get some info on the plugin
+   callDispatcher(tempHandle, effOpen, 0, 0, NULL, 0.0);
+
+   // Get the VST version the plugin understands
+   mVstVersion = callDispatcher(tempHandle, effGetVstVersion, 0, 0, NULL, 0);
+
+   // Ensure that it looks like a plugin and can deal with ProcessReplacing
+   // calls.  Also exclude synths for now.
+   if (   tempHandle->magic == kEffectMagic &&
+        !(tempHandle->flags & effFlagsIsSynth) &&
+          tempHandle->flags & effFlagsCanReplacing)
+   {
+      if (mVstVersion >= 2)
+      {
+         mName = GetStringFromHandle(effGetEffectName, tempHandle);
+         if (mName.length() == 0)
+         {
+            mName = GetStringFromHandle(effGetProductString, tempHandle);
+         }
+      }
+      if (mName.length() == 0)
+      {
+         mName = wxFileName{ mRealPath }.GetName();
+      }
+
+      if (mVstVersion >= 2)
+      {
+         mVendor = GetStringFromHandle(effGetVendorString, tempHandle);
+         mVersion = wxINT32_SWAP_ON_LE(callDispatcher(tempHandle, effGetVendorVersion, 0, 0, NULL, 0));
+      }
+      if (mVersion == 0)
+      {
+         mVersion = wxINT32_SWAP_ON_LE(tempHandle->version);
+      }
+
+      if (tempHandle->flags & effFlagsHasEditor || tempHandle->numParams != 0)
+      {
+         mInteractive = true;
+      }
+
+      mAudioIns = tempHandle->numInputs;
+      mAudioOuts = tempHandle->numOutputs;
+
+      mMidiIns = 0;
+      mMidiOuts = 0;
+
+      // Check to see if parameters can be automated.  This isn't a guarantee
+      // since it could be that the effect simply doesn't support the opcode.
+      mAutomatable = false;
+      for (int i = 0; i < tempHandle->numParams; i++)
+      {
+         if (callDispatcher(tempHandle, effCanBeAutomated, 0, i, NULL, 0.0))
+         {
+            mAutomatable = true;
+            break;
+         }
+      }
+   }
+   else
+   {
+      return false;
+   }
+
+   // close the temporary handle
+   callDispatcher(tempHandle, effClose, 0, 0, NULL, 0.0);
+
    return true;
 }
 
 
 bool VSTEffect::Load()
 {
-   // TODO: move this out of here, so that it can be called only once
+   // TODO: move this call out of here, so that it is executed only once
    if ( ! LoadCommon() )
       return false;
```

Comment:
Here's another idea.  Change the return type of `LoadCommon` to be the new structure.  Give it an `explicit operator bool()` returning false when it contains a null pointer, and return `{}` for failure.

Define move construction and assignment for the new type, to transfer the pointer to the left hand side and leave it null on the right.

Do all of this -- and the loading of the plug-in can avoid destroying one AEffect only to build another one at once.